### DISCUSSION TITLE
Add builder previews  with peek

### DISCUSSION THREAD

**Discussion by luckykenlin** [Z3d0X/filament-fabricator#194] [Ideas]
Hi everyone,

I recently discovered that Fabricator integrates very well with [filament-peek](https://github.com/pboivin/filament-peek).

how can I get this feature of [builder previews](https://github.com/pboivin/filament-peek/blob/2.x/docs/builder-previews.md#using-builder-previews-on-edit-pages).

Has anyone achieved this?

<img width="847" alt="image" src="https://github.com/user-attachments/assets/b3beeba1-e560-4f39-ae8c-aa3d19ab4c32" />

**Comment by pboivin** [comment]
Hi @luckykenlin,

I'm not aware of any project using Fabricator with Peek's builder preview. I think it could be done, but would require a small PR to Fabricator.

Looking at the Page Resource, it seems like the [`HasPreviewModal`](https://github.com/Z3d0X/filament-fabricator/blob/2.x/src/Resources/PageResource/Pages/Concerns/HasPreviewModal.php) trait would need to be modified to implement the [builder preview behavior](https://github.com/pboivin/filament-peek/blob/2.x/docs/builder-previews.md#using-builder-previews-on-edit-pages).

Other than that, you could probably use the [`blocks.before`](https://github.com/Z3d0X/filament-fabricator/blob/2.x/src/Resources/PageResource.php#L49) schema slot to add the `InlinePreviewAction`.
